### ISSUE TITLE
Fix release template issues/gaps (v0.11 edition)

### ISSUE THREAD

**Issue by BigLep** [ipfs/go-ipfs#8576]
### Checklist

- [X] This is a bug report, not a question. Ask questions on [discuss.ipfs.io](https://discuss.ipfs.io).
- [X] I have searched on the [issue tracker](https://github.com/ipfs/go-ipfs/issues?q=is%3Aissue) for my bug.
- [X] I am running the latest [go-ipfs version](https://dist.ipfs.io/#go-ipfs) or have an issue updating.

### Installation method

ipfs-desktop

**Comment by BigLep** [comment]
@galargh : before we started doing a PR per release for fixing up the release template, we had accumulated some improvements here.  I'm not sure how many of these are still relevant.  When you free up, can you please own going through this and seeing what is still relevant?  Thank yo.

**Comment by galargh** [comment]
I'll review it ahead of v0.18.0-RC1

**Comment by galargh** [comment]
> Address this gap: [Reach out to PL comms team after the initial release notes are drafted and then more closely the week of the release](https://protocollabs.slack.com/archives/C018EJ8LWH1/p1633718487175700)

This is not part of the release process. Is it obsolete now?

>  Remove the copy/pasted content in https://docs.ipfs.io/how-to/configure-node/

Do you know what this is referring to?

>  Have a single Discuss post for RC's so that interested testers can just watch the post for when we post new RCs. Template needs to be updated to reference that post in the "Invite the wider community through (link to the release issue)" section

We could do that. However, only the top level post is reposted to matrix, Slack, etc. So I wanted to confirm first if that's desired.

> Add Brave to the group of early testers so they get notified of a new version.

Working on this one in https://github.com/ipfs/kubo/pull/9487

>  Add a step for manually running the FUSE and Docker tests since they don't run in CI

I think we do run docker tests now. What's the reason for not `TEST_NO_FUSE=1` flag? https://github.com/ipfs/kubo/blob/master/.circleci/main.yml#L152

>  Review the package managers in https://github.com/ipfs/go-ipfs#native-linux-package-managers

In review: https://github.com/ipfs/kubo/pull/9488

>  Offload the communication side of releases (except for internal #ipfs)

I'm assuming bridges between different channels take care of this.

>  Add step for communicating with operator group about new RC

I'm assuming this means bifrost team.

>  Add a step reviewing infra config. Should be able to answer what to do based on release notes.

Do we want to review all of this - https://github.com/protocol/bifrost-infra/tree/master/ansible/inventories/bifrost/group_vars - during the release?

> Add a step for reviewing/updating dependency versions (could be running a go command, looking at dependabot PRs)

Do we want to do that during the release?

**Comment by BigLep** [comment]
@galargh : ACK that you have some questions here.  It's on my list to circle back this week.

**Comment by BigLep** [comment]
Thanks for seeing this through @galargh !

>> Address this gap: [Reach out to PL comms team after the initial release notes are drafted and then more closely the week of the release](https://protocollabs.slack.com/archives/C018EJ8LWH1/p1633718487175700)

> This is not part of the release process. Is it obsolete now?

Yeah, obsolete.  Let's strike it out.

>>  Remove the copy/pasted content in https://docs.ipfs.io/how-to/configure-node/

> Do you know what this is referring to?

No.  Let's strike it out.  

>>  Have a single Discuss post for RC's so that interested testers can just watch the post for when we post new RCs. Template needs to be updated to reference that post in the "Invite the wider community through (link to the release issue)" section

> We could do that. However, only the top level post is reposted to matrix, Slack, etc. So I wanted to confirm first if that's desired.

Good callout.  Is the way to satisfy all usecases to do a post per release and to have a long running post that someone can be subscribed to?  The long running post will just link to the individual release posts?

>>  Add a step for manually running the FUSE and Docker tests since they don't run in CI

> I think we do run docker tests now. What's the reason for not `TEST_NO_FUSE=1` flag? https://github.com/ipfs/kubo/blob/master/.circleci/main.yml#L152

We do need to run Docker tests - great that that is happening automatically. now!
Let's not worry about FUSE.  This isn't an area we're gong to invest in.  (An issue concerning deprecating it needs to be created, but that's a separate item).

>>  Offload the communication side of releases (except for internal #ipfs)

> I'm assuming bridges between different channels take care of this.

This was more about the marketing announcement, etc.  I think we have this weel covered and can call this done.

>>  Add step for communicating with operator group about new RC

> I'm assuming this means bifrost team.

This is about posting in the FIL Slack private #ipfs-operators channel.

>>  Add a step reviewing infra config. Should be able to answer what to do based on release notes.

> Do we want to review all of this - https://github.com/protocol/bifrost-infra/tree/master/ansible/inventories/bifrost/group_vars - during the release?

I think we want to make sure that the Kubo configuration being used in production makes sense to us and is giving us the kind of feedback we'd want about Kubo.  Ideally this gets to a state where we check to see "was there any new config options or changes?  If so, should we leverage them in Bifrost"?  I agree this will be harder to mechanize.  I would check to see if @guseggert has thoughts here.

>> Add a step for reviewing/updating dependency versions (could be running a go command, looking at dependabot PRs)

> Do we want to do that during the release?

I assume the concern here is the variability it will bring during release time. The main concern here is to make sure we have a mechanism where we're updating things.  Doing so after the release makes sense.  My only expectation is that it happens periodically and that it doesn't rely on best intentions.

**Comment by galargh** [comment]
> > > Add a step reviewing infra config. Should be able to answer what to do based on release notes.
> 
> > Do we want to review all of this - https://github.com/protocol/bifrost-infra/tree/master/ansible/inventories/bifrost/group_vars - during the release?
> 
> I think we want to make sure that the Kubo configuration being used in production makes sense to us and is giving us the kind of feedback we'd want about Kubo. Ideally this gets to a state where we check to see "was there any new config options or changes? If so, should we leverage them in Bifrost"? I agree this will be harder to mechanize. I would check to see if @guseggert has thoughts here.

I think maybe `What's left for release? > Required` is a good place for that. That way we'd look at it during standup and if anyone has any concerns, they can raise them there and turn into PRs. I added it to the 0.19 as an example: https://github.com/ipfs/kubo/issues/9502

**Comment by galargh** [comment]
I'm going to close this one now as we completed most of the items here. Things to consider in the future: modifying how we create Discuss posts during the release (if we do want to proceed with the changes in this area, let's create a new issue).
